include/public/marco/Codegen/Lowering/Distance/EditDistanceParameters.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MARCO_CODEGEN_LOWERING_DISTANCE_EDITDISTANCEPARAMETERS_H
+#define MARCO_CODEGEN_LOWERING_DISTANCE_EDITDISTANCEPARAMETERS_H
+
+#include <array>
+#include <cassert>
+#include <string>
+
+namespace marco::codegen::lowering
+{
+  // Given a character, calculate its index in the editDistanceSubsitutionCosts matrix.
+  // The only allowed characters are digits, letters and '_'.
+  inline unsigned int charToSubstitutionCostsIndex(const char &character) {
+    assert(character >= '0');
+    if (character <= '9') {
+      return character - '0';
+    }
+    assert(character >= 'A');
+    if (character <= 'Z') {
+      return character - 'A' + 10;
+    }
+    assert(character >= '_');
+    if (character == '_') {
+      return 36;
+    }
+    assert(character >= 'a' && character <= 'z');
+    return character - 'a' + 37;
+  }
+
+  // Costs for different edit operations.
+  constexpr unsigned int editDistanceBaseInsertionCost = 2;
+  constexpr unsigned int editDistanceDeletionCost = 2;
+  constexpr unsigned int editDistanceTranspositionCost = 3;
+  const unsigned int editDistanceLargestSubstitutionCost = 21;
+  const std::array<std::array<unsigned int, 63>, 63> editDistanceSubsitutionCosts =
```

Comment:
Where does this come from? I know the short answer, which is not sufficient. Other people don't even know that.

Reply:
I will explain it more carefully in the documentation